### PR TITLE
Documentation: add 2.2.1 changelog and fix build

### PR DESCRIPTION
**Pull request by penguinolog**
* Fix warning on `AsyncioEventLoop` docstring
* Add `/documentation` directory to `.gitignore`
* Add minimalistic tox target for documentation build

**NOTE: Official support for building Pages with Actions is in public beta at the moment.**
When it will be released officially, switch is expected

##### Checklist
- [ ] I've ensured that similar functionality has not already been implemented
- [ ] I've ensured that similar functionality has not earlier been proposed and declined
- [X] I've branched off the `master` or `python-dual-support` branch
- [X] I've merged fresh upstream into my branch recently
- [X] I've ran `tox` successfully in local environment
- [X] I've included docstrings and/or documentation and/or examples for my code (if this is a new feature)

### CODE REVIEW

**Comment by wardi** [comment]
parallel /docs and /documentation folders kind of make me sad. Isn't the sphinx default to build into docs/build or similar?

**Comment by penguinolog** [comment]
> parallel /docs and /documentation folders kind of make me sad. Isn't the sphinx default to build into docs/build or similar?

Current sphinx boilerplate uses:
`/doc/source` for documentation sources
`/doc/build` for built documentation

Project uses non-standard sphinx configuration, so we can set any folder name for built documentation.

**Comment by wardi** [comment]
I vote for anything that doesn't have two "docs" folders in the project root :-)

**Comment by penguinolog** [comment]
> I vote for anything that doesn't have two "docs" folders in the project root :-) 

Made as it implemented in actual boilerplate

File: .gitignore [hunk 1]
```diff
@@ -451,4 +451,7 @@ secring.*
 # Generated version
 /urwid/version.py
 # Built packages
-/wheelhouse
\ No newline at end of file
+/wheelhouse
+
+### Documentation build
+/docs/source
```
Comment (wardi) [inline .gitignore:457]: /docs/build?

Comment (wardi) [reply .gitignore:457]: and does the build still work from the new directory? I remember there were a few places in the docs that referenced the source, maybe those are broken by moving the files.

File: docs/source/conf.py [hunk 2]
```diff
@@ -48,8 +48,9 @@
 # |version| and |release|, also used in various other places throughout the
 # built documents.
 FILE_PATH = os.path.dirname(__file__)
-VERSION_MODULE = os.path.abspath(os.path.join(FILE_PATH,
-    '../urwid/version.py'))
+VERSION_MODULE = os.path.abspath(
+    os.path.join(FILE_PATH, '../../urwid/version.py')
```
Comment (penguinolog) [inline docs/source/conf.py:52]: Path changed due to hierarchy change

File: .gitignore (hunk 1 above)
Comment (penguinolog) [reply .gitignore]:
```
~/PycharmProjects/urwid (docs_build?) $ tox -e docs
.pkg: _optional_hooks> python /home/alekseis/.virtualenv/urwid/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: get_requires_for_build_editable> python /home/alekseis/.virtualenv/urwid/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
.pkg: build_editable> python /home/alekseis/.virtualenv/urwid/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
docs: install_package> python -I -m pip install --force-reinstall --no-deps /home/alekseis/PycharmProjects/urwid/.tox/.tmp/package/98/urwid-2.2.2.dev3+g4cde7ee-0.editable-cp311-cp311-linux_x86_64.whl
docs: commands[0]> sphinx-build docs/source docs/build
Running Sphinx v7.2.6
making output directory... done
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 29 source files that are out of date
updating environment: [new config] 29 added, 0 changed, 0 removed
reading sources... [100%] tutorial/index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
copying assets... copying static files... done
copying extra files... done
done
/home/alekseis/PycharmProjects/urwid/urwid/widget/widget_decoration.py:docstring of urwid.widget.widget_decoration.WidgetDecoration.selectable:10: WARNING: more than one target found for cross-reference 'keypress': urwid.Widget.keypress, urwid.Edit.keypress, urwid.IntEdit.keypress, urwid.Button.keypress, urwid.CheckBox.keypress, urwid.TreeWidget.keypress, urwid.SelectableIcon.keypress, urwid.Padding.keypress, urwid.Filler.keypress, urwid.Frame.keypress, urwid.ListBox.keypress, urwid.TreeListBox.keypress, urwid.Columns.keypress, urwid.Pile.keypress, urwid.GridFlow.keypress, urwid.Overlay.keypress
writing output... [100%] tutorial/index
/home/alekseis/PycharmProjects/urwid/urwid/widget/widget_decoration.py:docstring of urwid.widget.widget_decoration.WidgetDisable.selectable:10: WARNING: more than one target found for cross-reference 'keypress': urwid.Widget.keypress, urwid.Edit.keypress, urwid.IntEdit.keypress, urwid.Button.keypress, urwid.CheckBox.keypress, urwid.TreeWidget.keypress, urwid.SelectableIcon.keypress, urwid.Padding.keypress, urwid.Filler.keypress, urwid.Frame.keypress, urwid.ListBox.keypress, urwid.TreeListBox.keypress, urwid.Columns.keypress, urwid.Pile.keypress, urwid.GridFlow.keypress, urwid.Overlay.keypress
generating indices... genindex done
writing additional pages... index search done
copying images... [100%] tutorial/adventure4.png
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 2 warnings.

The HTML pages are in docs/build.
.pkg: _exit> python /home/alekseis/.virtualenv/urwid/lib/python3.11/site-packages/pyproject_api/_backend.py True setuptools.build_meta
  docs: OK (5.38=setup[0.86]+cmd[4.52] seconds)
  congratulations :) (5.44 seconds)
```

**Comment by penguinolog** [comment]
Rebased

File: .gitignore (hunk 1 above)
Comment (penguinolog) [reply .gitignore]: @wardi added documentation build pipeline, artifact with documentation will be available for 90 days after build.
Publishing for documentation should be done separately.

File: docs/source/conf.py (hunk 2 above)
Comment (wardi) [reply docs/source/conf.py:52]: this file doesn't exist anymore, does it?

**Comment by wardi** [comment]
I remember building the docs and comparing with the current version would sometimes catch bugs introduced into widgets or rendering code (because the example images would change)

WDYT about having CI build the docs and then report the diff between the built docs and the github-pages branch if there is any?

File: docs/source/conf.py (hunk 2 above)
Comment (penguinolog) [reply docs/source/conf.py:52]: it's generated during package build/install. It should be never added to git.

**Comment by penguinolog** [comment]
> I remember building the docs and comparing with the current version would sometimes catch bugs introduced into widgets or rendering code (because the example images would change)
> 
> WDYT about having CI build the docs and then report the diff between the built docs and the github-pages branch if there is any?

In theory possible, but need to think how to implement this on CI

**Comment by penguinolog** [comment]
Found better solution, than moving anything: `/build` is already used for built package, just generate directory with documentation and put it there

**Pull request merged**